specs/agents/span-links.md

```diff
@@ -0,0 +1,36 @@
+## Span Links
+
+A Span or Transaction MAY link to zero or more other Spans/Transactions that are causally related.
+
+Example use-cases for Span Links:
+
+1. When a single transaction represents the batch processing of several messages, the agent SHOULD be able to link back to the traces that have produced the messages.
+2. When the agent receives a `traceparent` header from outside a trust boundary, it SHOULD restart the trace (creating a different trace id with its own sampling decision) and link to the originating trace.
+3. Close gap for the OTLP intake - [OTel's specification of span links](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/overview.md#links-between-spans)
+
+Spans and Transactions MUST collect links in the `links` array with the following fields on each item:
```

Comment:
Should/must agents provide some (public) api to add links to the current transaction/span?

Reply:
Yes, the plan is to look into extending the public API as well, but that'll be a separate spec on top of this. This one is about how agents internally handle span links.

Issue for extending the public API: https://github.com/elastic/apm/issues/594

Reply:
@tobiasstadler I'd be happy to have opinions on the Java API for this at #594, especially because I'm not a Java guy. I'm drafting the spec there now. Briefly, I think the agent's API should be compatible with https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#specifying-links and current OTel language-specific API definitions, and in such a way that if/when we add support for span link *attributes*, they fit into the existing API smoothly.